### PR TITLE
Make sure to return an IPv4 SystemID

### LINKED ISSUE #1501: Test Error in Build in SystemIdTest.testNostId
**Describe the bug**

```
[ERROR] Failures: 
[ERROR]   SystemIdTest.testNostId:38 
Expecting actual:
  "fe80:0:0:0:aede:48ff:fe00:1122%en4"
to match pattern:
  "\d+\.\d+\.\d+\.\d+"
[INFO] 
[ERROR] Tests run: 100, Failures: 1, Errors: 0, Skipped: 3
```

**To Reproduce**
1. On macOS - run `mvn package`
2. Error happens in `nb-api` tests

**What was Expected**
Expect test to always pass.

### PR DESCRIPTION
**Pull request by dave2wave**
Fixes #1501 

If the code doesn't find an Ipv4 address the first time look deeper into the network interfaces

### CODE REVIEW

File: nb-api/src/main/java/io/nosqlbench/api/metadata/SystemId.java [hunk 1]
```diff
@@ -70,9 +72,21 @@ private static Optional<InetAddress> getMainInetAddrDirect() {
                 if (i2 instanceof Inet4Address) return 1;
                 return 0;
             }).findFirst();
-        return firstInetAddrForInterface;
-    }
 
+        if (firstInetAddrForInterface.orElse(null).getHostAddress().contains(".")) return firstInetAddrForInterface;
```
Comment (jshook) [inline nb-api/src/main/java/io/nosqlbench/api/metadata/SystemId.java:76]: Rephrase this line? This is a potential NPE

**Review by jshook — CHANGES_REQUESTED** [review]
one minor change requested

File: nb-api/src/main/java/io/nosqlbench/api/metadata/SystemId.java (hunk 1 above)
Comment (dave2wave) [reply nb-api/src/main/java/io/nosqlbench/api/metadata/SystemId.java]: fixed

**Review by jshook — APPROVED** [review]

File: nb-api/src/main/java/io/nosqlbench/api/metadata/SystemId.java [hunk 2]
```diff
@@ -70,9 +72,22 @@ private static Optional<InetAddress> getMainInetAddrDirect() {
                 if (i2 instanceof Inet4Address) return 1;
                 return 0;
             }).findFirst();
-        return firstInetAddrForInterface;
-    }
 
+        if ( firstInetAddrForInterface.isPresent() &&
+             firstInetAddrForInterface.orElse(null).getHostAddress().contains(".")) return firstInetAddrForInterface;
```
Comment (msmygit) [inline nb-api/src/main/java/io/nosqlbench/api/metadata/SystemId.java:77]: I know this PR is already merged, but do we really need the `orElse(null)` on here?

Comment (dave2wave) [reply nb-api/src/main/java/io/nosqlbench/api/metadata/SystemId.java:77]: Good point. I have a new PR almost ready to fix this.

**Pull request merged**